packages/angular/cli/src/commands/cache/info/cli.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { tags } from '@angular-devkit/core';
+import { promises as fs } from 'fs';
+import { join } from 'path';
+import { Argv } from 'yargs';
+import {
+  CommandModule,
+  CommandModuleImplementation,
+  CommandScope,
+} from '../../../command-builder/command-module';
+import { isCI } from '../../../utilities/environment-options';
+import { getCacheConfig } from '../utilities';
+
+export class CacheInfoCommandModule extends CommandModule implements CommandModuleImplementation {
+  command = 'info';
+  describe = 'Prints disk cache configuration and statistics in the console.';
+  longDescriptionPath?: string | undefined;
+  static override scope: CommandScope.In;
+
+  builder(localYargs: Argv): Argv {
+    return localYargs.strict();
+  }
+
+  async run(): Promise<void> {
+    const { path, environment, enabled } = getCacheConfig(this.context.workspace);
+
+    this.context.logger.info(tags.stripIndents`
+      Enabled: ${enabled ? 'yes' : 'no'}
+      Environment: ${environment}
+      Path: ${path}
+      Size on disk: ${await this.getSizeOfDirectory(path)}
+      Effective status on current machine: ${this.effectiveEnabledStatus()}
+    `);
+  }
+
+  private async getSizeOfDirectory(path: string): Promise<string> {
+    const directoriesStack = [path];
+    let size = 0;
+
+    while (directoriesStack.length) {
+      const dirPath = directoriesStack.pop();
+      if (dirPath === undefined) {
```

Comment:
Why is this necessary? Doesn't `while (directoriesStack.length)` already handle an empty stack? Is there a case where the stack is not empty but has an `undefined`?

Reply:
`pop()` has a return type of `T | undefined`.  The conditional was probably for TypeScript to not error.  Two alternatives would be a `while (true)` (downside of an extra pop call) or `const dirPath = directoriesStack.pop()!;` (downside of an eslint ignore comment).

Reply:
Indeed the conditional is for TypeScript. 

`while (true)` would still require an eslint ignore comment. So I'll go with a non null assertion.

Reply:
Ah gotcha, thanks for the clarification. Agreed that the assertion is at least clear that "This shouldn't happen, but we have to tell types that".